### PR TITLE
Nitter v2

### PR DESCRIPTION
**Pull request by zedeus**
Just in time to save us from #157, Nitter lives on to see another day.

Tons of changes, completely rewritten backend, many edgecases have been fixed and lots of effort has been put into making sure parsing and rendering is "correct". Redis is now used as a cache instead of SQLite, meaning you _must_ run a Redis instance for nitter. Host and port can be set in the config, along with some connection limits. There are some other new config options, but you likely don't need to change them, the defaults are fine.

Using Redis as a cache lets us cut out a lot of redundant requests, something Twitter's web client doesn't do. For example, fetching a timeline takes ~4-500 ms with nitter, while it takes upwards of 1.5 seconds with Twitter. Feel free to come talk to me on IRC about how the new backend works, if you have any questions, or general feedback less suited for a GitHub issue.

I apologize for not having this ready at least a few days ago for testing, but this took a ton of work since I started from scratch with reverse engineering the new undocumented API.

### CODE REVIEW

**Comment by B0pol** [comment]
I can't build it, I've followed steps from readme
`nimble build -d:release` is fine (I had to update nim, but that's not the point here)
`nimble scss` is also fine
`mkdir ./tmp` same
`./nitter` give me this error:
```
asyncfutures.nim(383)    read
asyncfutures.nim(383)    read
asyncfutures.nim(383)    read
asyncfutures.nim(383)    read
asyncfutures.nim(383)    read
Error: unhandled exception: Software caused connection abort [OSError]
```
I guess it is related to Redis as you mentioned "you _must_ run a Redis instance for nitter". Therefore could you update the README steps?

**Comment by zedeus** [comment]
Sure, haven't touched the readme yet

File: src/redis_cache.nim [hunk 1]
```diff
@@ -17,8 +17,12 @@ proc setCacheTimes*(cfg: Config) =
...
 
 proc initRedisPool*(cfg: Config) =
-  pool = waitFor newRedisPool(cfg.redisConns, maxConns=cfg.redisMaxConns,
-                              host=cfg.redisHost, port=cfg.redisPort)
+  try:
+    pool = waitFor newRedisPool(cfg.redisConns, maxConns=cfg.redisMaxConns,
+                                host=cfg.redisHost, port=cfg.redisPort)
+  except OSError:
+    echo "Failed to connect to Redis."
+    quit()
```
Comment (B0pol) [inline src/redis_cache.nim:25]: you should quit without a 0 return code if this error happened.

**Comment by B0pol** [comment]
I've found a regression, see #171

**Comment by B0pol** [comment]
I've found another regression, this time more important, see https://github.com/zedeus/nitter/issues/172

**Comment by B0pol** [comment]
Another remark:
I've seen you have changed preview of retweets, and I really like it! It's way more visible, and we can even play the video / go to the image url without needing to click on the tweet.
Though I think there is a room for improvement:
image
The space between the the user name, display name, date and the video is really small, that would be great to enlarge it.

I've also noticed you have removed the duration: 
image
Is it a limitation from new api?

**Pull request merged**